CONTRIBUTING.md

```diff
@@ -1,88 +1,203 @@
+  "shortName": "ts.WinUI.Proj.Default",
+  "identity": "ts.WinUI.Proj.Default",
+  "groupIdentity": "ts.WinUI.Proj.Default",
+  "description": "",
+  "tags": {
+    "language": "C#",
+    "type": "project",
+    "ts.type": "project",
+    "ts.projecttype": "Blank|SplitView|MenuBar",
+    "ts.frontendframework": "MVVMToolkit",
+    "ts.platform": "WinUI",
+    "ts.appmodel": "Desktop",
+    "ts.outputToParent": "true",
+    "ts.version": "1.0.0",
+    "ts.displayOrder": "0",
+    "ts.licenses": "[Microsoft.WindowsAppSDK](https://www.nuget.org/packages/Microsoft.WindowsAppSDK/1.0.1/License)"
+  },
+  ...
+}
+```
+
+The `ts.compositionFilter` indicates that the composition template will be applied to this base Project template when the conditions in the filter are met.
+
+Below are the fields that can be evaluated in a composition filter:
+
+* template.json fields
+  * `name`
+  * `identity`
+  * `groupIdentity`
+  * any `tag`
+* context parameters
+  * `$frontendframework`
+  * `$projectType`
+
+Conditionals can be combined with the `&` operator and fields can be compared to literal values with the `==` and `!=` operators.
+
+New files can be contributed to the base Project template by adding them to the composition template following the folder structure of the base Project template.
+
+Existing files can be modified with [merge post actions](https://github.com/microsoft/CoreTemplateStudio/blob/dev/docs/templates.md#merge-post-action). Merge post actions are what allow you to modify existing files in the template. For example, if you want to add or remove properties from the base Project template .csproj file, you can do so in the composition template with a merge post action applied to the .csproj file.
+
+Merge post actions on a file are defined by adding a file to the composition template with the same name as the original file but with a _postaction suffix before the file extension (e.g. `Param_ProjectName_postaction.csproj`).
+
+Merge post action files use pattern matching and special comments to identify a location within the original file and then either add or remove content at that location.
+
+```xml
+<Project Sdk="Microsoft.NET.Sdk">
+  <PropertyGroup>
+    <EnablePreviewMsixTooling>true</EnablePreviewMsixTooling>
+<!--{--{-->
+    <WindowsPackageType>None</WindowsPackageType>
+<!--}--}-->
+  </PropertyGroup>
+```
+The above merge post action file would search for the `EnablePreviewMsixTooling` property and then remove `<WindowsPackageType>None</WindowsPackageType>` found after that location by surrounding the line to remove with XML comments that use the `{--{` and `}--}` syntax. This syntax indicates that the content within the comment should be removed from the original file. See https://github.com/microsoft/CoreTemplateStudio/blob/dev/docs/templates.md#post-actions for more details on merge post action syntax as well as other types of post actions that can change the output after generation has occurred.
+
+### Modifying the Wizard
```

Comment:
This section seems to be more about adding new template options within the wizard, rather than modifying the wizard itself.

We probably need sections for both.  - Make this about adding new entries within the wizard and create a follow-up issue for me to add a section about modifying the actual wizard.

Reply:
Yes the intent of this section is how to control what templates show up in the wizard and how to control relationships between templates. This is likely the majority case for community contributions.

Will create a separate issue for modifying the wizard code itself.

Reply:
#4468